Description: Apply fixes used in previous Pi releases
  Fix for certain wifi dongles not being detected; use clientid rather than duid
 .
 dhcpcd5 (6.11.3-1+rpi1) jessie; urgency=medium
 .
   * Merged in changes from 6.9.4-1+rpi2
Author: Simon Long <simon@raspberrypi.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- dhcpcd5-6.11.3.orig/dhcpcd.conf
+++ dhcpcd5-6.11.3/dhcpcd.conf
@@ -8,12 +8,12 @@
 hostname
 
 # Use the hardware address of the interface for the Client ID.
-#clientid
+clientid
 # or
 # Use the same DUID + IAID as set in DHCPv6 for DHCPv4 ClientID as per RFC4361.
 # Some non-RFC compliant DHCP servers do not reply with this set.
 # In this case, comment out duid and enable clientid above.
-duid
+#duid
 
 # Persist interface configuration when dhcpcd exits.
 persistent
--- dhcpcd5-6.11.3.orig/if.c
+++ dhcpcd5-6.11.3/if.c
@@ -508,7 +508,7 @@ if_discover(struct dhcpcd_ctx *ctx, int
 		/* We reserve the 100 range for virtual interfaces, if and when
 		 * we can work them out. */
 		ifp->metric = 200 + ifp->index;
-		if (if_getssid(ifp) != -1) {
+		if (if_getssid(ifp) != -1 || *ifp->name == 'w') {
 			ifp->wireless = 1;
 			ifp->metric += 100;
 		}
